### PR TITLE
[IOTDB-3088]Implement MTreeAboveSG serialize and deserialize

### PR DESCRIPTION
**Pull request by MarcosZyk**
## Description

Implement MTreeAboveSG serialize and deserialize for confignode usage.

### CODE REVIEW

File: server/src/test/java/org/apache/iotdb/db/metadata/mtree/MTreeAboveSGTest.java [hunk 1]
```diff
@@ -289,4 +292,49 @@ public void testGetNodeListInLevel() throws MetadataException {
...
+      storageGroupMNode.setTimePartitionInterval(i);
+    }
+
+    ByteBuffer byteBuffer = ByteBuffer.allocate(1024 * 1024);
+    root.serialize(byteBuffer);
+    byteBuffer.flip();
+
+    root.clear();
+
+    root.deserialize(byteBuffer);
```
Comment (CRZbulabula) [inline server/src/test/java/org/apache/iotdb/db/metadata/mtree/MTreeAboveSGTest.java:321]: How about new a MTreeAboveSG, then do deserialization for it? This is how the ConfigNode use it~

**Review by CRZbulabula — COMMENTED** [review]
PTAL~

File: server/src/test/java/org/apache/iotdb/db/metadata/mtree/MTreeAboveSGTest.java (hunk 1 above)
Comment (MarcosZyk) [reply server/src/test/java/org/apache/iotdb/db/metadata/mtree/MTreeAboveSGTest.java]: Fixed.

**Review by CRZbulabula — APPROVED** [review]
LGTM~

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**